core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/score/director/ScoreDirectorFactoryFactory.java

```diff
@@ -70,9 +71,45 @@ protected AbstractScoreDirectorFactory<Solution_, Score_> decideMultipleScoreDir
                 ServiceLoader.load(ScoreDirectorFactoryService.class);
         Map<ScoreDirectorType, Supplier<AbstractScoreDirectorFactory<Solution_, Score_>>> scoreDirectorFactorySupplierMap =
                 new EnumMap<>(ScoreDirectorType.class);
+        boolean isBavet = false;
```

Comment:
Maybe use the `ConstraintStreamImplType` enum here (and make it comparable/has a priority)? Alternatively, iterate through a sorted version of `scoreDirectorFactoryServiceLoader`'s elements (like what you do in the test).

Reply:
There is a difference here to what I do in the test.

In the test, I *know* `cs-common` is on the classpath, therefore I can use its facilities to tell if BAVET was loaded, or if it was DROOLS. Here in `core`, I do not have that luxury, and therefore I have to accomplish the same thing in this ugly roundabout way.